app/controllers/api/v1/book_lists_controller.rb

```diff
@@ -0,0 +1,58 @@
+class Api::V1::BookListsController < ApplicationController
+  def index
+    render json: BookList.all
+  end
+
+  def create
+    book_list = BookList.create book_list_params
+    if book_list
+      render json: { message: "List created." }, status: :ok
+    else
+      render json: { message: "Error" }, status: :error
+    end
+  end
+
+  def add
+    book = Book.find(params[:book_id])
+    errors = []
+    params[:booklist_ids].each do |id|
+      booklist = BookList.find(id)
+      if booklist.books.include?(book)
+        errors << booklist.name
+      else
+        booklist.books << book
+      end
+    end
+    if errors.empty?
+      render json: { message: "Book successfully added" }, status: :ok
+    else
+      render json: { errors: errors }, status: :error
+    end
+  end
+
+  def books
+    render json: BookList.find(params[:id]).books
+  end
+
+  def csv
+    send_data Book.to_csv(BookList.find(params[:id]).books),
+      :type => 'text/csv; charset=iso-8859-1; header=present',
```

Comment:
Align the parameters of a method call if they span more than one line.<br>Use the new Ruby 1.9 hash syntax.<br>Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.